src/shogun2-web/src/main/java/de/terrestris/shogun2/rest/GroupRestController.java

```diff
@@ -0,0 +1,76 @@
+package de.terrestris.shogun2.rest;
+
+import java.util.Map;
+import java.util.Set;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import de.terrestris.shogun2.dao.UserGroupDao;
+import de.terrestris.shogun2.model.User;
+import de.terrestris.shogun2.model.UserGroup;
+import de.terrestris.shogun2.service.UserGroupService;
+import de.terrestris.shogun2.util.data.ResultSet;
+
+/**
+ * @author Johannes Weskamm
+ * @author Nils Bühner
+ *
+ */
+@RestController
+@RequestMapping("/groups")
+public class GroupRestController<E extends UserGroup, D extends UserGroupDao<E>, S extends UserGroupService<E, D>>
```

Comment:
This should be named `UserGroupRestController` as our Model/Dao/Service is also using `UserGroup`.